File: Sources/Grovs/model/entities/AppDetails.swift
//
//  AppDetails.swift
//
//  grovs
//

/// A structure representing details of the application.
struct AppDetails: Codable {
    let version: String
    let build: String
    let bundle: String
    let device: String
    let deviceID: String
    let userAgent: String
    let screenWidth: Int?
    let screenHeight: Int?
    let language: String?
    let timeZone: String?

    /// Coding keys for decoding and encoding.
    private enum CodingKeys : String, CodingKey {
        case version, build, bundle, device, deviceID, userAgent, screenWidth, screenHeight, language, timeZone
    }

    /// Converts the app details to a backend-compatible dictionary.
    ///
    /// - Returns: A dictionary containing the app details.
    func toBackend() -> [String: Any] {
        return [
            "user_agent": userAgent,
            "vendor_id": deviceID,
            "app_version": version,
            "build": build,
            "device": device,
            "timezone": timeZone,
            "language": language,
            "screen_width": screenWidth,
            "screen_height": screenHeight
        ]
    }
}

File: sdks/android/Grovs/Grovs/src/main/java/io/grovs/model/AppDetails.kt
package io.grovs.model

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class AppDetails (
    @SerializedName("app_version")
    var version: String,
    var build: String,
    var bundle: String,
    var device: String,
    @SerializedName("vendor_id")
    var deviceID: String,
    @SerializedName("user_agent")
    var userAgent: String,
    var url: String? = null,
    @SerializedName("screen_width")
    var screenWidth: String? = null,
    @SerializedName("screen_height")
    var screenHeight: String? = null,
    var timezone: String? = null,
    var language: String? = null,
    @SerializedName("webgl_vendor")
    var webglVendor: String? = null,
    @SerializedName("webgl_renderer")
    var webglRenderer: String? = null,
) : Parcelable {
}